tests: internal: signv4: fixed leak (CID 508222)

leonardo-albertovich opened the pull request.

edsiper commented:
@leonardo-albertovich note this PR is changing Fluent Bit version

![image](https://github.com/user-attachments/assets/5c79c0ae-37f9-46a0-a9e8-8169b08852f1)

edsiper commented:
wrong dup of https://github.com/fluent/fluent-bit/pull/9345

leonardo-albertovich commented:
That's because I changed the base branch as Pat taught me today but did it backwards. 
I reopened this one and switched it back to master because it's still valid.
